communication/nats/connection_mock.go

```diff
@@ -101,6 +104,9 @@ func (conn *ConnectionMock) MessageWait(waitChannel chan interface{}) (interface
 
 // Publish publishes a new message
 func (conn *ConnectionMock) Publish(subject string, payload []byte) error {
+	conn.m.Lock()
```

Comment:
No point to lock here as lock not used

Reply:
I experienced data races related to connection_mock, perhaps this fixed it.

Reply:
Yes, it addresses a race in tests. Reduced a scope a bit.